pyanaconda/modules/storage/kickstart.py

```diff
@@ -144,6 +146,51 @@ def parse(self, args):
         return fc
 
 
+class Iscsi(COMMANDS.Iscsi):
+    def parse(self, args):
+        tg = super().parse(args)
+
+        if tg.iface:
+            if not wait_for_network_devices([tg.iface]):
+                raise KickstartParseError(lineno=self.lineno,
+                        msg=_("Network interface \"%(nic)s\" required by iSCSI \"%(iscsiTarget)s\" target is not up.") %
+                             {"nic": tg.iface, "iscsiTarget": tg.target})
+
+        mode = iscsi.mode
+        if mode == "none":
+            if tg.iface:
+                network_proxy = NETWORK.get_proxy()
+                activated_ifaces = network_proxy.GetActivatedInterfaces()
+                iscsi.create_interfaces(activated_ifaces)
+        elif ((mode == "bind" and not tg.iface)
+              or (mode == "default" and tg.iface)):
+            raise KickstartParseError(lineno=self.lineno,
+                    msg=_("iscsi --iface must be specified (binding used) either for all targets or for none"))
+
+        try:
+            if tg.target:
+                log.info("adding iscsi target %s at %s:%d via %s", tg.target, tg.ipaddr, tg.port, tg.iface)
+            else:
+                log.info("adding all iscsi targets discovered at %s:%d via %s", tg.ipaddr, tg.port, tg.iface)
+            iscsi.add_target(tg.ipaddr, tg.port, tg.user,
+                             tg.password, tg.user_in,
+                             tg.password_in,
+                             target=tg.target,
+                             iface=tg.iface)
+        except (IOError, ValueError) as e:
+            raise KickstartParseError(lineno=self.lineno, msg=str(e))
+
+        return tg
+
+
+class IscsiName(COMMANDS.IscsiName):
+    def parse(self, args):
+        retval = super().parse(args)
+
+        iscsi.initiator = self.iscsiname
```

Comment:
I guess that the initiator is required by the `iscsi` commands to work. Otherwise, it should be set only by the `set_initiator` method of the ISCSI module.

Reply:
Yes, it is.

Reply:
But I wonder - shouldn't we move all the iscsi actions (attaching targets) from parse() of Iscsi (and IscsiName but it is already there - setting the initiator name) commands to process_kickstart?

Reply:
I was thinking about it and I don't like that the DBus method ReadKickstart can cause so many "side effects". Eventually, we could just return a list of DBus tasks that would be created based on the parsed kickstart and the boss could run these "actions" directly. It would be great for testing as well, because right now we have to prevent most of these actions in unit tests.

However, the current approach is good enough for now. I have created an issue for it, so we can revisit it later.

Reply:
Yes, now that we don't care about connecting the devices before processing (parsing) ignoredisk --onlyuse <iscsi_disk> we can do it with tasks in a cleaner way.